### ISSUE TITLE
Reader::createFromPath does not release instance of the file for removal

### ISSUE THREAD

**Issue by youvtr** [thephpleague/csv#229]
## Issue summary

_(Please explain in plain english your issue/feature request)_

Reader::createFromPath does not release instance ff the file for removal

### System informations

_(In case of a bug report Please complete the table below)_

| Information | Description |
|--------------|---------|
| League\Csv version | 8.0.0  |
| PHP/HHVM version |  7.0.10 |
| OS Platform |  windows 7|

## Standalone code, or other way to reproduce the problem

_(Please complete the text below to help us fix the issue)_

$fileUploadedPath =  '/path/to/file.csv';
$reader = Reader::createFromPath( $fileUploadedPath );
// do some stuff with csv from reader api
unlink( $fileUploadedPath );

### Expected result
File should be removed from folder

### Actual result
File still present on folder and log  unlink(/path/to/file.csv): Permission denied  

### Notes
folders configured with chmod 777

**Comment by nyamsprod** [comment]
@youvtr that's because you still have a reference to the file in your `$reader` instance yous should do this instead.

```php
 <?php

$fileUploadedPath = '/path/to/file.csv';
$reader = Reader::createFromPath( $fileUploadedPath );
// do some stuff with csv from reader api
$reader = null;  //this line make sure that the internal reference to the file is removed
unlink( $fileUploadedPath ); //unlink should work now.
```

This is a PHP limitation not a CSV library bug.

**Comment by youvtr** [comment]
@nyamsprod Thanks for your help but even after implementing your suggestion the same error happens.

**Comment by nyamsprod** [comment]
Then it only means that somewhere else in your script you still have a resource or a reference present. But it is not a League\Csv issue per se. Please refer to #173
